### PR TITLE
Add Most of the User Routes

### PR DESCRIPTION
**Pull request by NicoClack**
- Add: GET by id, POST and DELETE user
- Dependency: Zod
- Fix: process.env.DATABASE_URL support

To do later:

- GET /v1/users/:id/articles
- PUT /v1/users/:id/bio
- PUT /v1/users/:id/password ?

[Trello](https://trello.com/c/NcXi95TG)

### CODE REVIEW

**Review by BlueElyas — APPROVED** [review]
A damn fine job

**Review by kamila-wilczynska — APPROVED** [review]

File: backend/src/routes/users_routes.js [hunk 1]
```diff
@@ -14,4 +15,97 @@ userRouter.get("/", async (req, res, next) => {
...
+// 	try {
+// 		const result = pool.query(
+// 			"SELECT users.username, articles.title, articles.description, articles.date_published FROM users JOIN articles ON users.id = $1::number",
+// 			[id],
+// 		);
+// 	} catch (err) {
+// 		console.log(err);
+// 		res.status(500).send("Server error :(");
+// 	}
+// });
```
Comment (Sebbybobbler) [inline backend/src/routes/users_routes.js]: Purposefully left in?

Comment (BlueElyas) [reply backend/src/routes/users_routes.js]: 👀👀👀

**Review by Sebbybobbler — APPROVED** [review]
Now THIS I love!

**Pull request merged**